pkg/operation/botanist/controlplane.go

```diff
@@ -400,25 +400,48 @@ func (b *Botanist) DeploySeedMonitoring() error {
 		values["alertmanager"].(map[string]interface{})["emailConfigs"] = emailConfigs
 	}
 
-	return b.ApplyChartSeed(filepath.Join(common.ChartPath, "seed-monitoring"), fmt.Sprintf("%s-monitoring", b.Shoot.SeedNamespace), b.Shoot.SeedNamespace, nil, values)
+	if err := b.ApplyChartSeed(filepath.Join(common.ChartPath, "seed-monitoring"), fmt.Sprintf("%s-monitoring", b.Shoot.SeedNamespace), b.Shoot.SeedNamespace, nil, values); err != nil {
+		return err
+	}
+
+	// If no mail receivers for the Shoot alerts provided than there is no need
+	// to run the Alertmanager and it can be removed from the control plane.
+	if len(emailConfigs) == 0 {
+		if err := b.deleteSeedAlertmanager(); err != nil && !apierrors.IsNotFound(err) {
+			return err
+		}
+	}
+	return nil
 }
 
 // DeleteSeedMonitoring will delete the monitoring stack from the Seed cluster to avoid phantom alerts
 // during the deletion process. More precisely, the Alertmanager and Prometheus StatefulSets will be
 // deleted.
 func (b *Botanist) DeleteSeedMonitoring() error {
-	err := b.K8sSeedClient.DeleteStatefulSet(b.Shoot.SeedNamespace, common.AlertManagerStatefulSetName)
-	if apierrors.IsNotFound(err) {
-		return nil
+	if err := b.deleteSeedPrometheus(); err != nil {
+		return err
 	}
-	if err != nil {
+	if err := b.deleteSeedAlertmanager(); err != nil {
 		return err
 	}
-	err = b.K8sSeedClient.DeleteStatefulSet(b.Shoot.SeedNamespace, common.PrometheusStatefulSetName)
-	if apierrors.IsNotFound(err) {
-		return nil
+	return nil
+}
+
+func (b *Botanist) deleteSeedPrometheus() error {
+	if err := b.K8sSeedClient.DeleteStatefulSet(b.Shoot.SeedNamespace, common.PrometheusStatefulSetName); err != nil && !apierrors.IsNotFound(err) {
+		return err
 	}
-	return err
+	return nil
+}
+
+func (b *Botanist) deleteSeedAlertmanager() error {
+	if err := b.K8sSeedClient.DeleteStatefulSet(b.Shoot.SeedNamespace, common.AlertManagerStatefulSetName); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
Any reason why not ALL resources are cleaned up that belong to the alertmanager?
Previously, this function was only executed during shoot deletion, so here we did only delete the STS because we knew that the remaining resources will go soon anyways.
However, now this is part of the normal shoot reconciliation, so I guess it would make sense to clean everything up (see https://github.com/gardener/gardener/blob/2d0e57a7b77e4c904ceeb765ac2179fda39436d4/pkg/operation/common/utils.go#L442).